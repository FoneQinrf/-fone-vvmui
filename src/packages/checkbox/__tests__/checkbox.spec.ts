/*
 * @Author: Fone`峰
 * @Date: 2021-04-22 11:27:20
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-05-06 10:46:14
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount, shallowMount } from '@vue/test-utils';
import Checkbox from '../index.vue';
import CheckboxGroup from '../../checkbox-group/index.vue';
import { nextTick } from 'vue';

describe('Checkbox', () => {
  it('选中逻辑正常', async () => {
    const wrapper = shallowMount(CheckboxGroup, {
      slots: {
        default: [Checkbox, Checkbox]
      }
    });
    // await nextTick();
    // const find = wrapper.findComponent({ name: 'Checkbox' });
    // console.log(find);
  });
});
