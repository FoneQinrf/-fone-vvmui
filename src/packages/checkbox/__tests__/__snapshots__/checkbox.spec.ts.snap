// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render textTheme and label corretly 1`] = `"<label class=\\"vvm-checkbox --error --text-theme\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px;\\"></i></transition-stub></span><span class=\\"vvm-checkbox--label\\">checkbox</span></label>"`;

exports[`Checkbox should render type and textTheme corretly 1`] = `"<div class=\\"vvm-checkboxgroup --inline\\"><label class=\\"vvm-checkbox --success --text-theme\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px;\\"></i></transition-stub></span><span class=\\"vvm-checkbox--label\\">checkbox-1</span></label><label class=\\"vvm-checkbox --success\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span><span class=\\"vvm-checkbox--label\\">checkbox-2</span></label><label class=\\"vvm-checkbox --success\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span><span class=\\"vvm-checkbox--label\\">checkbox-3</span></label></div>"`;

exports[`Checkbox should render type corretly 1`] = `
"<label class=\\"vvm-checkbox --default\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span>
  <!--v-if-->
</label>"
`;

exports[`Checkbox should render type corretly 2`] = `
"<label class=\\"vvm-checkbox --success\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span>
  <!--v-if-->
</label>"
`;

exports[`Checkbox should render type corretly 3`] = `
"<label class=\\"vvm-checkbox --warning\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span>
  <!--v-if-->
</label>"
`;

exports[`Checkbox should render type corretly 4`] = `
"<label class=\\"vvm-checkbox --error\\"><span class=\\"vvm-checkbox--icon vvm-ignore\\"><transition-stub><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconroundcheckfill --checkbox-icon\\" style=\\"font-size: 24px; width: 24px; height: 24px; display: none;\\"></i></transition-stub></span>
  <!--v-if-->
</label>"
`;
